pkg/chaosdaemon/client/utils.go

```diff
@@ -41,16 +47,55 @@ func (c *GrpcChaosDaemonClient) Close() error {
 	return c.conn.Close()
 }
 
+func FindDaemonIP(ctx context.Context, c client.Client, pod *v1.Pod) (string, error) {
+	nodeName := pod.Spec.NodeName
+	log.Info("Creating client to chaos-daemon", "node", nodeName)
+
+	ns := config.ControllerCfg.Namespace
+	var endpoints v1.Endpoints
+	err := c.Get(ctx, types.NamespacedName{
+		Namespace: ns,
+		Name:      "chaos-daemon",
+	}, &endpoints)
+	if err != nil {
+		return "", err
+	}
+
+	daemonIP := findIPOnEndpoints(&endpoints, nodeName)
+	if len(daemonIP) == 0 {
+		return "", errors.Errorf("cannot find daemonIP on node %s in related Endpoints %v", nodeName, endpoints)
+	}
+
+	return daemonIP, nil
+}
+
+func findIPOnEndpoints(e *v1.Endpoints, nodeName string) string {
+	for _, subset := range e.Subsets {
+		for _, addr := range subset.Addresses {
+			if addr.NodeName != nil && *addr.NodeName == nodeName {
+				return addr.IP
+			}
+		}
+	}
+
+	return ""
+}
+
 // NewChaosDaemonClient would create ChaosDaemonClient
-func NewChaosDaemonClient(ctx context.Context, c client.Client, pod *v1.Pod, port int) (ChaosDaemonClientInterface, error) {
+func NewChaosDaemonClient(ctx context.Context, c client.Client, pod *v1.Pod) (ChaosDaemonClientInterface, error) {
```

Comment:
We need a little refactoring with `NewChaosDaemonClient` and `NewChaosDaemonClientLocally`:

This file is under package `pkg/chaosdaemon/client`, it should be the minimal client constructor of the chaos-daemon, so I do not think kuernetes and chaos-controller-manager things should be here.

I think we could:

- remove these two methods
- create a constructor of `GrpcChaosDaemonClient`
- If we need the origin `NewChaosDaemonClient`, we could write a similar one at `pkg/controllers`